pizza_cutter/des_pizza_cutter/_coadd_slices.py

```diff
@@ -555,3 +567,132 @@ def _coadd_slice_inputs(
+def _get_gaia_stars(*, fname):
+    """
+    load the gaia stars
+
+    Parameters
+    -----------
+    fname: str
+        path to the gaia star catalog
+    """
+    full_path = os.path.expandvars(fname)
+    logger.info('reading: %s' % full_path)
+    data = fitsio.read(full_path, lower=True)
+    data = eu.numpy_util.add_fields(data, [('radius_arcsec', 'f4')])
+
+    ply = np.poly1d([0.00443223, -0.22569131, 2.99642999])
+
+    # these do not have a radius factor in them
+    log10_radius_arcsec = ply(data['phot_g_mean_mag'])
+    data['radius_arcsec'] = 10.0**log10_radius_arcsec
+    return data
+
+
+@njit
+def _intersects(row, col, radius_pixels, nrows, ncols):
+    """
+    low level routine to check if the mask intersects the image.
+    For simplicty just check the bounding rectangle
+
+    Parameters
+    ----------
+    row, col: float
+        The row and column of the mask center
+    radius_pixels: float
+        The radius for the star mask
+    nrows, ncols: int
+        Shape of the image
+
+    Returns
+    -------
+    True if it intersects, otherwise False
+    """
+
+    low_row = -radius_pixels
+    high_row = nrows + radius_pixels - 1
+    low_col = -radius_pixels
+    high_col = ncols + radius_pixels - 1
+
+    if (
+        row > low_row and row < high_row and
+        col > low_col and col < high_col
+    ):
+        return True
+    else:
+        return False
+
+
+@njit
+def _do_mask_gaia_stars(rows, cols, radius_pixels, bmask, flag):
+    """
+    low level code to mask stars
+
+    Parameters
+    ----------
+    rows, cols: arrays
+        Arrays of rows/cols of star locations; may be off the image
+    radius_pixels: float
+        The radius for each star mask
+    bmask: array
+        The bmask to modify
+    flag: int
+        The flag value to "or" into the bmask
+    """
+    nrows, ncols = bmask.shape
+
+    for istar in range(rows.size):
+        row = rows[istar]
+        col = cols[istar]
+        rad = radius_pixels[istar]
+        rad2 = rad * rad
+
+        if not _intersects(row, col, rad, nrows, ncols):
+            continue
+
+        for irow in range(nrows):
+            rowdiff2 = (row - irow)**2
+            for icol in range(ncols):
+
+                r2 = rowdiff2 + (col - icol)**2
+                if r2 < rad2:
+                    bmask[irow, icol] |= flag
+
+
+def _mask_gaia_stars(gaia_stars, wcs, bmask, radius_factor=1, maxmag=18):
+    """
+    mask gaia stars, setting a bit in the bmask
+
+    Parameters
+    ----------
+    gaia_stars: array
+        Array with fields ra, dec, radius_arcsec
+    wcs: WCS object
+        The WCS object for the image
+    bmask: array
+        The bmask to modify
+    radius_factor: float, optional
+        Factor by which to multiply the radius.  Default 1.0
+    maxmag: float, optional
+        Maximum mag to mask.  Default 18
+    """
+    from ._se_image import _compute_wcs_area
+
+    pixel_scale = np.sqrt(_compute_wcs_area(wcs, 100, 100))
```

Comment:
We should use the WCS at the patch center. This can be computed from start_row, start_col and the box_size parameters in the calling function.

Reply:
I thought the caodd wcs was a constant pixel scale

Reply:
It has a very small amount of distortion since it is a curved sky WCS. I use the actual WCS from the DES coadd tile.

Reply:
OK.  Getting the pixel scale wrong by a percent doesn't matter.  But it is certainly better to use a calculated location rather than 100